packages/frontend/src/retro/components/buttons/ToggleRetroTimerButton.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from "react";
+import { Alarm, SnoozeOutlined } from "@mui/icons-material";
+import { ActionButton } from "../../../common/components/buttons/ActionButton";
+import { isModerator } from "../../../common/utils/participantsUtils";
+import { CreateTimerDialog } from "../dialogs/CreateTimerDialog";
+
+import { useUserContext } from "../../../common/context/UserContext";
+import { useDialog } from "../../../common/hooks/useDialog";
+import { useTimer } from "../../hooks/useTimer";
+
+import MaibornConfetti from "../../../common/components/MaibornConfetti";
+
+export function ToggleRetroTimerButton() {
+  const { isOpen, closeDialog, openDialog } = useDialog();
+  const { user } = useUserContext();
+  const [isParticlesActive, setIsParticlesActive] = useState(false);
+
+  const confettiDuration = 10000;
+
+  const {
+    minutes,
+    seconds,
+    remainingTimeLabel,
+    startTimer,
+    stopTimer,
+    pauseTimer,
+    isTimerRunning,
+    isTimerPaused,
+  } = useTimer({
+    onTimerFinish: handleTimerFinish,
+    runtime: 300000,
+  });
```

Comment:
I think we should move this to the context and extract that information from there. The remainingTime needs to be tracked outside of the context though